### ISSUE TITLE
wl_subcompositor issue

### ISSUE THREAD

**Issue by levone1** [elros34/qxcompositor#2]
Hi. Would you mind taking a look here - https://github.com/Smithay/client-toolkit/issues/210

I'm using qxcompositor from this repo to run Devuan desktop on SailfishOS.

Thanks

**Comment by elros34** [comment]
With my little knowledge this info from Readme.md "OpenGL terminal emulator" is main blocker. You only have OpenGl ES on your smartphone.
Have you tried to run it under X, if that is even possible?
You can safely ignore whole qxcompositor and run alacritty under lipstick to see if that change anything. QXCompositor is really not needed for tests. Another idea: run it under weston or even better kwin_wayland inside Devuan. Kwin should be quite usable, and have both wayland and xwayland support and probably all latest wayland protocols enabled and supported including wl_subcompositor.

libhyris acceleration could also be important otherwise even if terminal would work, there would be no any kind of acceleration. In sailfish_linux_chroot I handle it: https://github.com/elros34/sailfish_linux_chroot/blob/master/ubuntu/scripts/install.sh#L49 but no idea if harbour containers also have it.

Which version of qxcompositor do you run? Latest do not have SubSurfaceExtension enabled which could be what you want because nested compsitor is really used to handle orientation correctly and xwayland: https://github.com/elros34/qxcompositor/blob/qtwayland-5.6/src/qmlcompositor.cpp#L63. Even full support for qt applications in wayland mode is not ready. In my test subsurfaces caused many issues with current qtwayland we use so I do not handle them correctly: https://github.com/elros34/qxcompositor/blob/qtwayland-5.6/src/qmlcompositor.cpp#L168.

In short qxcompositor is not really full compositor like kwin, weston or sway. It's just window rotator for lipstick.

**Comment by levone1** [comment]
> With my little knowledge this info from Readme.md "OpenGL terminal emulator" is main blocker. You only have OpenGl ES on your smartphone. Have you tried to run it under X, if that is even possible? You can safely ignore whole qxcompositor and run alacritty under lipstick to see if that change anything. QXCompositor is really not needed for tests. Another idea: run it under weston or even better kwin_wayland inside Devuan. Kwin should be quite usable, and have both wayland and xwayland support and probably all latest wayland protocols enabled and supported including wl_subcompositor.
> 
> libhyris acceleration could also be important otherwise even if terminal would work, there would be no any kind of acceleration. In sailfish_linux_chroot I handle it: https://github.com/elros34/sailfish_linux_chroot/blob/master/ubuntu/scripts/install.sh#L49 but no idea if harbour containers also have it.
> 
> Which version of qxcompositor do you run? Latest do not have SubSurfaceExtension enabled which could be what you want because nested compsitor is really used to handle orientation correctly and xwayland: https://github.com/elros34/qxcompositor/blob/qtwayland-5.6/src/qmlcompositor.cpp#L63. Even full support for qt applications in wayland mode is not ready. In my test subsurfaces caused many issues with current qtwayland we use so I do not handle them correctly: https://github.com/elros34/qxcompositor/blob/qtwayland-5.6/src/qmlcompositor.cpp#L168.
> 
> In short qxcompositor is not really full compositor like kwin, weston or sway. It's just window rotator for lipstick.

Thanks... I'm using whatever version comes with "git clone..."

would it be easy to enable subsurfaceextension in source, just to test? I guess the panic simply happens if theres no value for wl_subcompositor,  so if there was a value, that would clear. I don't know if that means it will then work, but...

**Comment by elros34** [comment]
git clone will use master branch which is latest for sfos < 4.2 but not for sfos >= 4.2 which must use qtwayland-5.6 branch otherwise qxcompositor will not compile or start at all. 
Adding SubsurfaceExtension is trivial, just look at the line I mentioned previously.
